pkg/controller/deployment/progress_test.go

```diff
@@ -352,3 +352,103 @@ func TestSyncRolloutStatus(t *testing.T) {
 		})
 	}
 }
+
+func getRSConditions(typeConditionsReplicas []apps.ReplicaSetConditionType, statusConditionsReplicas []v1.ConditionStatus) []apps.ReplicaSetCondition {
+	var conditions []apps.ReplicaSetCondition
+	for i := 0; i < len(typeConditionsReplicas); i++ {
+		Condition := apps.ReplicaSetCondition{
+			Type:   typeConditionsReplicas[i],
+			Status: statusConditionsReplicas[i],
+		}
+		conditions = append(conditions, Condition)
+	}
+	return conditions
+}
+
+func newRSWithCondition(name string, specReplicas int, typeConditionsReplicas []apps.ReplicaSetConditionType, statusConditionsReplicas []v1.ConditionStatus) *apps.ReplicaSet {
+	rs := rs(name, specReplicas, nil, metav1.Time{})
+	rs.Status.Conditions = getRSConditions(typeConditionsReplicas, statusConditionsReplicas)
+	return rs
+}
+
+func TestGetReplicaFailures(t *testing.T) {
+	typeConditionsReplicasAll := []apps.ReplicaSetConditionType{
+		apps.ReplicaSetReplicaFailure,
+		apps.ReplicaSetReplicaFailure,
+		apps.ReplicaSetReplicaFailure,
+	}
```

Comment:
I think that it could be useful to add some values different from `apps.ReplicaSetReplicaFailure` in this array and other similar arrays, to check that only `apps.ReplicaSetReplicaFailure` values are taken into account.